packages/backend-function/src/function_env_translator.ts

```diff
@@ -88,31 +88,29 @@ export class FunctionEnvironmentTranslator {
    * @param value envVar value that can be a plain text or a secret
    */
   addEnvironmentEntry = (key: string, value: string | BackendSecret) => {
-    {
-      if (key === this.amplifySsmEnvConfigKey) {
-        throw new Error(
-          `${this.amplifySsmEnvConfigKey} is a reserved environment variable name`,
-        );
-      }
-      if (typeof value === 'undefined') {
-        throw new AmplifyUserError('InvalidFunctionConfigurationError', {
-          message: `The value of environment variable ${key} is undefined.`,
-          resolution: `Ensure that all defineFunction environment variables are defined.`,
-        });
-      } else if (typeof value === 'string') {
-        this.lambda.addEnvironment(key, value);
-      } else {
-        const { branchSecretPath, sharedSecretPath } =
-          this.backendSecretResolver.resolvePath(value);
-        this.lambda.addEnvironment(key, this.ssmValuePlaceholderText);
-        this.ssmEnvVars[branchSecretPath] = {
-          name: key,
-          sharedPath: sharedSecretPath,
-        };
-        this.ssmPaths.push(branchSecretPath, sharedSecretPath);
-      }
-      this.amplifyBackendEnvVarNames.push(key);
+    if (key === amplifySsmEnvConfigKey) {
+      throw new Error(
+        `${amplifySsmEnvConfigKey} is a reserved environment variable name`,
+      );
+    }
+    if (typeof value === 'undefined') {
+      throw new AmplifyUserError('InvalidFunctionConfigurationError', {
+        message: `The value of environment variable ${key} is undefined.`,
+        resolution: `Ensure that all defineFunction environment variables are defined.`,
+      });
+    } else if (typeof value === 'string') {
+      this.lambda.addEnvironment(key, value);
+    } else {
+      const { branchSecretPath, sharedSecretPath } =
+        this.backendSecretResolver.resolvePath(value);
+      this.lambda.addEnvironment(key, ssmValuePlaceholderText);
+      this.ssmEnvVars[key] = {
+        path: branchSecretPath,
+        sharedPath: sharedSecretPath,
+      };
```

Comment:
Is there any possibility of breaking existing customers who might be relying on the shape of this env var?